src/plugins/intel_cpu/src/nodes/fullyconnected.cpp

```diff
@@ -922,6 +992,75 @@ bool FullyConnected::useSparseWeightsDecompression() {
 
     return true;
 }
+
+#ifdef OV_CPU_WITH_MLAS
+void FullyConnected::prepackMLASWeight() {
+    auto prepareMLASWeight = [&](const int64_t N, const int64_t K) {
+        if (!getParentEdgeAt(WEIGHTS_ID)->getParent()->isConstant())
+            IE_THROW() << "Weight input is not const for node " << getName() << ".";
+        auto weightsMem = getParentEdgeAt(WEIGHTS_ID)->getMemoryPtr();
+        std::string format = "gemm_mlas_" + std::to_string(N) + "_" + std::to_string(K);
+        if (!weightsMem)
+            IE_THROW() << "Cannot get const weights edgeMem for node " << getName() << ".";
+        auto packedBsize = ov_sgemm_pack_get_size(N, K);
+        MemoryPtr ptr;
+        auto create = [&]() {
+            float* weightPtr = reinterpret_cast<float*>(weightsMem->GetPtr());
+            size_t ldb = K;
+            MemoryPtr _ptr = std::make_shared<Memory>(getEngine());
+            _ptr->Create(
+                intel_cpu::CpuBlockedMemoryDesc(Precision::I8, intel_cpu::Shape{packedBsize}));
+            float* prepackedDst = reinterpret_cast<float*>(_ptr->GetPtr());
+            ov_sgemm_pack("T", N, K, ldb, weightPtr, prepackedDst);
+            return _ptr;
+        };
+
+        auto weightCache = context->getWeightsCache();
+        if (weightCache != nullptr) {
+            const std::string string_hash = getName() + "_" + format + "_" + std::to_string(weightsMem->GetSize()) +
+                                            "_" + std::to_string(reinterpret_cast<uint64_t>(weightsMem->GetData()));
+
+            ptr = *weightCache->findOrCreate(string_hash, create);
+        } else {
+            ptr = create();
+        }
+        return ptr;
+    };
+    const auto& wgtDims = getParentEdgeAt(WEIGHTS_ID)->getMemoryPtr()->getStaticDims();
+    // Weight is transpoed by MatMulConstTransposesExtraction
+    // K is the IC of weight
+    K = wgtDims[1];
```

Comment:
Are we sure that weights always have second rank?

Reply:
I think so. 
The  `convert_to_matmul_to_fc` transformation has reshaped the weight to `[-1, K]` in Line 138